reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -877,12 +899,71 @@ OSMTurnRelation createTurnRelation(ReaderRelation relation) {
                 }
             }
             if (fromWayID >= 0 && toWayID >= 0 && viaNodeID >= 0) {
-                return new OSMTurnRelation(fromWayID, viaNodeID, toWayID, type);
+                return new OSMTurnRelation(fromWayID, viaNodeID, toWayID, type, vehicleTypeRestricted, vehicleTypesExcept);
```

Comment:
I am wondering: Do we need to do any kind of trimming for `vehicleTypeRestricted` and `vehicleTypesExcept` values (i.e. can there be whitespace before or after the tags in xml ? Also I wonder if we should keep the constructor with the first four arguments for `OSMTurnRelation` and add `setVehicleTypeRestricted`/`addVehicleTypesExcept` so we can get rid of some of the null checks ?

Reply:
I added trim function and try to remove the unneeded tests.